pkg/controller/volume/attachdetach/cache/desired_state_of_world.go

```diff
@@ -392,16 +395,18 @@ func (dsw *desiredStateOfWorld) GetVolumesToAttach() []VolumeToAttach {
 
 	volumesToAttach := make([]VolumeToAttach, 0 /* len */, len(dsw.nodesManaged) /* cap */)
 	for nodeName, nodeObj := range dsw.nodesManaged {
-		for volumeName, volumeObj := range nodeObj.volumesToAttach {
-			volumesToAttach = append(volumesToAttach,
-				VolumeToAttach{
-					VolumeToAttach: operationexecutor.VolumeToAttach{
-						MultiAttachErrorReported: volumeObj.multiAttachErrorReported,
-						VolumeName:               volumeName,
-						VolumeSpec:               volumeObj.spec,
-						NodeName:                 nodeName,
-						ScheduledPods:            getPodsFromMap(volumeObj.scheduledPods),
-					}})
+		if utilfeature.DefaultFeatureGate.Enabled(features.NodeShutdown) && !nodeObj.isShutdown {
```

Comment:
this does not work like this? imo remove featuregate from this.

Reply:
True updated.